### PR TITLE
Switch from PhantomJS to Chrome Headless via Puppeteer

### LINKED ISSUE #257: PhantomJS -> Chrome Headless
PhantomJS is [no longer being maintained](https://groups.google.com/d/msg/phantomjs/9aI5d-LDuNE/5Z3SMZrqAQAJ). Would the maintainers of this project be open to switching to headless chrome? I'd be open to submitting a PR. Any thoughts/direction?

### PR DESCRIPTION
**Pull request by stdavis**
Closes #257

This PR tears out PhantomJS (no longer maintained) and replaces it with Chrome Headless via the [Puppeteer](https://github.com/GoogleChrome/puppeteer) project. Our office got tired of fighting with PhamtomJS.

Here are the highlights of the changes:
- [x] Removed all references to phantomjs (not quite there yet)
- [x] Upgrade from jshint to eslint (required for the async/await code IIRC)
- [x] Cleaned up a bunch of apparently unused jasmine files
- [x] Upgraded min node version from 0.10.0 to 7.6.0 (required for async/await)

Let me and/or @steveoh know if you think that this has a chance to be merged and what we can do to make that happen.

Thanks for the great project!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,25 +9,25 @@
   "repository": "gruntjs/grunt-contrib-jasmine",
   "license": "MIT",
   "engines": {
-    "node": ">=0.10.0"
+    "node": ">=7.6.0"
```
[the remark this answers is not in the document]
Comment (steveoh) [reply package.json]: Wellll......

> Caution: Puppeteer requires at least Node v6.4.0, but the examples below use async/await which is only supported in Node v7.6.0 or greater.

[the remark this answers is not in the document]
Comment (vladikoff) [reply package.json]: Ah I see what happened, the ci files were auto generated from `internal`. So by adding this we would probably say that this supports node >=8  only

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# grunt-contrib-jasmine v1.1.0 [![Build Status: Linux](https://travis-ci.org/gruntjs/grunt-contrib-jasmine.svg?branch=master)](https://travis-ci.org/gruntjs/grunt-contrib-jasmine) [![Build Status: Windows](https://ci.appveyor.com/api/projects/status/5985958by5rhnh31/branch/master?svg=true)](https://ci.appveyor.com/project/gruntjs/grunt-contrib-jasmine/branch/master)
+# grunt-contrib-jasmine v1.2.0 [![Build Status: Linux](https://travis-ci.org/gruntjs/grunt-contrib-jasmine.svg?branch=master)](https://travis-ci.org/gruntjs/grunt-contrib-jasmine) [![Build Status: Windows](https://ci.appveyor.com/api/projects/status/5985958by5rhnh31/branch/master?svg=true)](https://ci.appveyor.com/project/gruntjs/grunt-contrib-jasmine/branch/master)
```
[the remark this answers is not in the document]
Comment (steveoh) [reply README.md]: It is a major change but it doesn’t break the current api.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (steveoh) [reply package.json]: Ok

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (steveoh) [reply README.md]: I'll revert this and let the versioning get handled by the merging folks.

**Comment by steveoh** [comment]
@vladikoff do you have any ideas for the `The http2 module is an experimental API`? It's failing in the current v8 master branch for the same reason.

**Comment by stdavis** [comment]
I was able to get the travis build running for our fork's repo: https://travis-ci.org/agrc/grunt-contrib-jasmine

I can't think of anything that would be different other than the cache. Perhaps you want to manually clear the cache in travis for this project?

As for the AppVeyor failure, it may be related to this: https://github.com/gruntjs/grunt-contrib-connect/issues/235. Not sure if you want to add the environment variable workaround or something different.

**Comment by steveoh** [comment]
Yeah, @vladikoff maybe try deleting the [build caches](https://travis-ci.org/gruntjs/grunt-contrib-jasmine/caches)

**Comment by stdavis** [comment]
I think that I have all referenced to PhantomJS replaced with Headless Chrome.

**Pull request merged**